chainer/functions/math/clip.py

```diff
@@ -55,7 +57,9 @@ def check_type_forward(self, in_types):
         type_check.expect(in_types[0].dtype.kind == 'f')
 
     def forward_cpu(self, inputs):
-        return utils.force_array(inputs[0] * self.cond),
+        gy, = inputs
+        gx = utils.force_array(gy * self.cond, gy.dtype)
+        return gx,
```

Comment:
Do we need the fix to `ClipGrad`?
```
>>> import numpy as np
>>> np.__version__
'1.17.0'
>>> np.result_type(np.float16, bool)
dtype('float16')
```

Reply:
Oh, it seems this part is not required.
But I think it's better to be explicit.

Reply:
`self.cond` should be boolean tensor so type conversion won't happen?
Then I like this expression more.